addons/api/addon/serializers/auth-method.js

```diff
@@ -17,18 +17,32 @@ export default class AuthMethodSerializer extends ApplicationSerializer {
    */
   serialize(snapshot) {
     switch (snapshot.record.type) {
-      case TYPE_AUTH_METHOD_OIDC:
-        return this.serializeOIDC(...arguments);
+      case TYPE_AUTH_METHOD_PASSWORD:
+        return this.serializePassword(...arguments);
       default:
         return this.serializeDefault(...arguments);
     }
   }
 
+  serializeAttribute(snapshot, json, key, attribute) {
+    super.serializeAttribute(...arguments);
+    const { options } = attribute;
+
+    // If an attribute has a `for` option, it must match the
+    // record's `type`, else the attribute is excluded
+    // from serialization.
+    if (options?.for && !options.for.includes('oidc', 'ldap')) {
+      if (options.isNestedAttribute) {
+        delete json.attributes[key];
+      }
+    }
+  }
```

Comment:
```suggestion
  serializeAttribute(snapshot, json, key, attribute) {
    const { type } = snapshot.record;
    super.serializeAttribute(...arguments);
    const { options } = attribute;

    // If an attribute has a `for` option, it must match the
    // record's `type`, else the attribute is excluded
    // from serialization.
    if (options?.for && !options?.for?.includes(type)) {
      if (options?.isNestedAttribute) {
        delete json?.attributes?.[key];
      }
    }
  }
```

Reply:
can you check if this works? 

the test failure was due to an empty attributes object which gets deleted in `serializeDefault`.